### PR TITLE
kube rebase 20160211

### PR DESCRIPTION
**Pull request by pweil-**
- [x] compile
- [x] test-go
- [x] test-integration
- [x] test-cmd
- [x] test-end-to-end
- [x] test proxy flag - confirmed config flag still works to switch proxy modes in `RunProxy`

Of Note:

1.  Admission control now uses a generated client.  ~~SCC is not in that interface and I was unable to get it generated in the fork with `hack/update-codegen.sh`~~.  Follow up for next rebase: https://github.com/openshift/origin/issues/7383

@soltysh @deads2k

### CODE REVIEW

**Comment by pweil-** [comment]
@soltysh @deads2k - now building and the vast majority of unit tests are passing

There are still some failures in

1.  a missing kubectl option in oc
1.  a controller test due to the new client pieces
1.  a round trip test 

Two issues I found this evening

1.  The bump for influxdb was not correct, I fixed it and squashed it in k8s bump commit here but the fork will need to be updated
1.  I hacked in the SCC interface to the generated client so I could continue addressing the refactoring and tests since I was unable to get it generated.

**Comment by deads2k** [comment]
@pmorie I've seen a pull from you where you spoke generated client.  Any advice for @pweil- ?

**Comment by deads2k** [comment]
We're carrying a patch on etcd: https://github.com/openshift/origin/pull/7259

**Comment by deads2k** [comment]
You're good to squash down the boring minus the quorum read.

**Comment by deads2k** [comment]
Looking good.

**Comment by pweil-** [comment]
squashed boring commits

**Comment by pweil-** [comment]
@deads2k - FYI, cherry-picking our patch to etcd (`ff1fe62187269cc5da3760de29dc03ae3698ff55`) resulted in an empty change set

**Comment by pweil-** [comment]
unit tests pass locally, moving on to integration and e2e

**Comment by liggitt** [comment]
when you bump etcd, did you do a merge, or an overwrite? we probably want to overwrite and reapply carries

**Comment by pweil-** [comment]
> when you bumped etcd, did you do a merge, or an overwrite? we probably want to overwrite and reapply carries

I guess it merged since the change set is here and not upstream.  I'll have to fix it.

**Comment by pweil-** [comment]
clean test-integration run done

**Comment by soltysh** [comment]
`rebase-20120211` ain't like living in the past, huh ;) ?

**Comment by pweil-** [comment]
> rebase-20120211 ain't like living in the past, huh ;) ?

lol

**Comment by pweil-** [comment]
[test]

**Comment by pweil-** [comment]
issue I'm seeing on e2e:

```
INFO] Validating rsync
Error: unknown command "rsh" for "kubectl"

Did you mean this?
	run

Run 'kubectl --help' for usage.
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: error in rsync protocol data stream (code 12) at io.c(226) [sender=3.1.1]
error: exit status 12
!!! Error in hack/../test/end-to-end/core.sh:262
	'oc rsync examples/sample-app ${frontend_pod}:/tmp' exited with status 1
Call stack:
	1: hack/../test/end-to-end/core.sh:262 main(...)
Exiting with status 1
!!! Error in hack/test-end-to-end.sh:65
	'${OS_ROOT}/test/end-to-end/core.sh' exited with status 1
Call stack:
	1: hack/test-end-to-end.sh:65 main(...)
Exiting with status 1
```

**Comment by pweil-** [comment]
rsync issue identified: https://github.com/kubernetes/kubernetes/issues/21316

**Comment by pweil-** [comment]
re[test]

**Comment by deads2k** [comment]
re[test]  This should show a pass before we stop the queue

**Comment by deads2k** [comment]
Are we still using non-generated generated clients?

**Comment by deads2k** [comment]
This looks pretty good.  You should pick the `UPSTREAM`s since your starting bump, carry them into your kube branch, then figure out how to bring that forward to here.  Let me know how you do that last step.  I had difficulty.

Once the commits are in good shape and we have one green test (not the whole job, just one), we can ask to stop the queue.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -170,128 +170,128 @@
...
+			"Comment": "v2.2.5",
+			"Rev": "bc9ddf260115d2680191c46977ae72b837785472"
 		},
 		{
 			"ImportPath": "github.com/coreos/etcd/pkg/idutil",
-			"Comment": "v2.2.2-4-ge0c7768",
-			"Rev": "e0c7768f94cdc268b2fce31ada1dea823f11f505"
+			"Comment": "v2.2.5",
+			"Rev": "bc9ddf260115d2680191c46977ae72b837785472"
 		},
```
Comment (miminar) [inline Godeps/Godeps.json]: Why don't we use the same version as k8s (which is fa94270d4bac0d8ae5dc6b71894e251aada93f74)?

I had to replace the `coreos/pkg` locally with upstream version of godep to make it compile. Details in https://github.com/pweil-/origin/pull/24

Comment (deads2k) [inline Godeps/Godeps.json]: @miminar you need to check your environment.  This builds without issues.

Comment (deads2k) [inline Godeps/Godeps.json]: Also, @miminar this is the level that kube is using for "github.com/coreos/etcd/pkg/ioutil". It looks like this bump is using matching level for things like capns log (fa94270d4bac0d8ae5dc6b71894e251aada93f74).  Can you double check your comments against the 62e574341c8e9c2136b23b8feffadea4c156b284 level of kube and your build environment.  Sounds like you have a local problem.

@

**Comment by pweil-** [comment]
[test]

**Comment by pweil-** [comment]
failure was:

```
F0217 09:00:42.570395    7108 node.go:164] unable to aquire file lock on "/var/run/lock/kubelet.lock": open /var/run/lock/kubelet.lock: permission denied
```

https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/11081/consoleFull#127309647156c25e74e4b05b79524e5d56

File: Godeps/Godeps.json (hunk 1 above)
Comment (miminar) [inline Godeps/Godeps.json]: Sorry for the false alarm. My environment was dirty. Running `rm -rf Godeps/_workspace/src && git checkout Godeps/_workspace` solved the problem. Still I wonder why untracked files under `Godeps/` are not listed in `git status`.

**Comment by pweil-** [comment]
@deads2k - the file lock was a new addition in the bump.  It is flagged as `<Warning: Alpha feature>` in the kubelet command line options.  I can default it to empty on our configuration.

**Comment by deads2k** [comment]
@pweil- that's a known problem that was "fixed" improperly: https://github.com/kubernetes/kubernetes/pull/21084 and https://github.com/kubernetes/kubernetes/issues/21078

@derekwaynecarr @liggitt @smarterclayton they side-stepped it by special casing launch conditions in their script.  now we have to decide what we're going to do.  A lock file makes no sense for us.  Are we going to patch it out or side-step it when launching?

**Comment by pweil-** [comment]
Last commit just defaults our config to off for now.

**Comment by liggitt** [comment]
It's defaulted to empty in https://github.com/kubernetes/kubernetes/pull/21089/files

**Comment by pweil-** [comment]
> It's defaulted to empty in https://github.com/kubernetes/kubernetes/pull/21089/files

you can pick your friends, you can pick your pulls, and you CAN pick your friend's pulls!  Picking!

**Comment by pweil-** [comment]
re[test]

**Comment by pweil-** [comment]
Cancelled last jenkins run - re[test]

**Comment by pweil-** [comment]
@liggitt cast your terrible gaze upon this PR

**Comment by jeremyeder** [comment]
@pweil- terrible _dead_ gaze. /cc @deadliggitt

**Comment by pweil-** [comment]
Green run: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/11103/

Changes since:

* squash of testing commit

**Comment by soltysh** [comment]
Except for the conversion stuff, I'm not sure about this LGTM.

**Comment by deads2k** [comment]
@danmcp let's just down the queue.  We've had one green test and the code won't change much from here.

**Comment by deads2k** [comment]
lgtm [merge] (in case the queue isn't deep)

**Comment by smarterclayton** [comment]
Can't merge this until #7184 lands

**Comment by pweil-** [comment]
> Can't merge this until #7184 lands

it just landed

**Comment by smarterclayton** [comment]
Now you can [merge]

**Comment by pweil-** [comment]
rebased for good measure

**Comment by deads2k** [comment]
[vipmerge]

**Pull request merged**